### PR TITLE
nerd-fonts: correct typos in descriptions

### PR DESCRIPTION
**Pull request by Ben9986**
Characters such as `&` and `'` were encoded as `&amp;` and `\u2019`. This fixes that.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by misuzu** [comment]
To properly fix the issue it's required to fix the [update script](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/nerd-fonts/update.py) or your changes will be overwritten by the next run of this script.

**Comment by Ben9986** [comment]
> To properly fix the issue it's required to fix the [update script](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/nerd-fonts/update.py) or your changes will be overwritten by the next run of this script.

ah missed that thanks, will have a look

**Comment by Ben9986** [comment]
So I've fixed the issue with apostrophes but I'm not good enough with python to fix the ampersand issue. Perhaps @rc-zb could attempt? I assume it would be trivial if I actually knew python 👀

**Comment by rc-zb** [comment]
No, I seldom write in Python and know little about it.

From search engine and LLM I heard [`html.unescape`](https://docs.python.org/3/library/html.html#html.unescape), and it seems to work.

File: pkgs/data/fonts/nerd-fonts/update.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def fetchjson(url):
 
 def storejson(path, obj):
     with open(path, "w", encoding="utf-8") as f:
-        json.dump(obj, f, indent=2)
+        json.dump(obj, f, indent=2, ensure_ascii = False)
```
Comment (misuzu) [inline pkgs/data/fonts/nerd-fonts/update.py:31]:
```suggestion
        json.dump(obj, f, indent=2, ensure_ascii=False)
```

**Comment by misuzu** [comment]
Try this:
```
def slicedict(d, ks):
    return {k: html.unescape(d[k]) for k in ks}
```
Make sure to import `html`

**Comment by Ben9986** [comment]
> Try this:
> 
> ```
> def slicedict(d, ks):
>     return {k: html.unescape(d[k]) for k in ks}
> ```
> 
> Make sure to import `html`

That's sorted it, thanks. 

Ready for merge imo

**Review by misuzu — APPROVED** [review]

**Pull request merged**